samples/cpp_synchronization/src/main.cpp

```diff
@@ -155,5 +155,5 @@ int main(void)
 		sem_main.wait();
 	}
 
-	return 0;
+	return;
```

Comment:
not necessary return in these cases, just remove them. Lets be concise with all others `void` functions in our code base.